File: magellan-library/src/main/java/com/wealthfront/magellan/ScreenContainer.java
package com.wealthfront.magellan;

import android.content.Context;
import android.os.Parcelable;
import android.util.AttributeSet;
import android.util.SparseArray;
import android.view.MotionEvent;
import android.widget.FrameLayout;

/**
 * The container to be used to display the screens using the {@link com.wealthfront.magellan.navigation.Navigator}. Must have the id
 * {@code magellan_container}. This will also block touch events automatically during navigation to avoid accidental
 * double taps.
 */
public class ScreenContainer extends FrameLayout {

  private boolean interceptTouchEvents;

  public ScreenContainer(Context context) {
    super(context);
  }

  public ScreenContainer(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  public ScreenContainer(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
  }

  @Override
  public boolean onInterceptTouchEvent(MotionEvent ev) {
    return interceptTouchEvents;
  }

  public void setInterceptTouchEvents(boolean interceptTouchEvents) {
    this.interceptTouchEvents = interceptTouchEvents;
  }

  @Override
  protected void dispatchRestoreInstanceState(
      SparseArray<Parcelable> container) {
    dispatchThawSelfOnly(container);
  }

  @Override
  protected void dispatchSaveInstanceState(SparseArray<Parcelable> container) {
    dispatchFreezeSelfOnly(container);
  }
}

File: magellan-kotlin/src/main/java/com/wealthfront/magellan/kotlin/ScreenContainer.kt
package com.wealthfront.magellan.kotlin

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.FrameLayout

/**
 * The container to be used to display the screens using the [Navigator]. Must have the id
 * `magellan_container`. This will also block touch events automatically during navigation to avoid accidental
 * double taps.
 */
class ScreenContainer : FrameLayout {

  private var interceptTouchEvents: Boolean = false

  constructor(context: Context) : super(context) {}

  constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

  constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

  override fun onInterceptTouchEvent(ev: MotionEvent): Boolean = interceptTouchEvents

  fun setInterceptTouchEvents(interceptTouchEvents: Boolean) {
    this.interceptTouchEvents = interceptTouchEvents
  }

}